components/camel-univocity-parsers/src/main/java/org/apache/camel/dataformat/univocity/Marshaller.java

```diff
@@ -86,7 +88,9 @@ private void writeRow(Exchange exchange, Object row, W writer) throws NoTypeConv
         Map<?, ?> map = convertToMandatoryType(exchange, Map.class, row);
         if (adaptHeaders) {
             for (Object key : map.keySet()) {
-                headers.add(convertToMandatoryType(exchange, String.class, key));

```

Comment:
@onders86 yeah it's hard to tell the original intent of that `headers` field, but could be that the unit tests also made assumptions about multi-threaded use. Anyways thank you for looking into this and fixing it 👍